src/cocotb/triggers.py

```diff
@@ -445,9 +445,15 @@ def __init__(self, parent: "Event") -> None:
         self._parent = parent
 
     def _prime(self, callback: Callable[[Trigger], None]) -> None:
-        self._callback = callback
-        self._parent._prime_trigger(self, callback)
-        super()._prime(callback)
+        if not self._primed:
```

Comment:
Nit: I'd return early here on `if self._primed` to reduce indentation, but either style is fine.